packages/openapi-generator/src/document-converter.ts

```diff
@@ -0,0 +1,84 @@
+import { promises } from 'fs';
+import { parse } from 'path';
+import { OpenAPIV3 } from 'openapi-types';
+import { convert } from 'swagger2openapi';
+import { safeLoad } from 'js-yaml';
+import { errorWithCause } from '@sap-cloud-sdk/util';
+const { readFile } = promises;
+
+/**
+ * Convert an OpenApi document to ensure smooth parsing and generation thereafter.
+ * Documents are expected to be formatted as JSON, OpenApi version 3 and only have one "default" tag.
+ * @param filePath File content of the original spec.
+ * @param ouputFilePath Path to write the altered spec to.
+ */
+export async function convertOpenApiSpec(
+  filePath: string
+): Promise<OpenAPIV3.Document> {
+  let file;
+  try {
+    file = await parseFileAsJson(filePath);
+  } catch (err) {
+    throw errorWithCause('Could not parse OpenApi specification.', err);
+  }
+  const openApiDocument = await convertDocToOpenApiV3(file);
```

Comment:
Maybe we catch potential errors here so we understand it fails when converting from v2.0 to v3.0?

Reply:
Done.